### PR TITLE
Cache terms

### PR DESCRIPTION
**Pull request by aliak00**

### CODE REVIEW

File: Source/Manager/SDKConfiguration/SDKConfiguration+AgreementsCache.swift [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        /// Maximum number of days agreements status should be cached
+        public let maxDays: UInt32
+
+        func load(forUserID userID: String) -> Agreements? {
+            guard self.isOn, let data = Settings.value(forKey: Keys.agreementsData) as? JSONObject else {
+                return nil
+            }
+            do {
+                let storedID = try data.string(for: Keys.Data.userID)
+                guard storedID == userID else {
```
Comment (Torsph) [inline Source/Manager/SDKConfiguration/SDKConfiguration+AgreementsCache.swift]: Any reason to not combine the `guards` here? One test on each line?

File: Source/Manager/SDKConfiguration/SDKConfiguration+AgreementsCache.swift [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        }
+
+        func store(_ agreements: Agreements, forUserID userID: String) {
+            guard self.isOn else {
+                return
+            }
+            let now = Date()
+            let hoursPerDay: UInt32 = 24
+            let number = Int(arc4random_uniform(UInt32(self.maxDays - self.minDays) * hoursPerDay) + self.minDays * hoursPerDay)
+            guard let later = Calendar.current.date(byAdding: .hour, value: number, to: now) else {
```
Comment (Torsph) [inline Source/Manager/SDKConfiguration/SDKConfiguration+AgreementsCache.swift]: later dude 🦆

Comment (Torsph) [inline Source/Manager/SDKConfiguration/SDKConfiguration+AgreementsCache.swift]: Smart to use the Calendar functions todo this 👍 No need to think about day wrapper, daylight saving etc.

File: Source/Manager/SDKConfiguration/SDKConfiguration+AgreementsCache.swift (hunk 1 above)
Comment (aliak00) [reply Source/Manager/SDKConfiguration/SDKConfiguration+AgreementsCache.swift]: Just readability. Also might need try? but then I can't log the throw if it throws

Comment (Torsph) [reply Source/Manager/SDKConfiguration/SDKConfiguration+AgreementsCache.swift]: The try-catch is ok.. Was thinking more about the 3 guards :)

**Pull request merged**